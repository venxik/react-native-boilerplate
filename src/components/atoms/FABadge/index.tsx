import React, { memo } from 'react';

import { Badge } from 'native-base';

import type { IBadgeProps } from 'native-base';

export type IFABadgeProps = IBadgeProps;

export const FABadge = memo((props: IFABadgeProps) => {
  const { testID, ...baseProps } = props;

  return <Badge testID={testID} {...baseProps} />;
});
FABadge.displayName = 'FABadge';
