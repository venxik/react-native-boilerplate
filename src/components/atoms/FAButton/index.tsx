import React, { memo } from 'react';

import { Button } from 'native-base';

import type { IButtonProps } from 'native-base';

export type IFAButtonProps = IButtonProps;

export const FAButton = memo((props: IFAButtonProps) => {
  const { testID, ...baseProps } = props;

  return <Button testID={testID} {...baseProps} />;
});
FAButton.displayName = 'FAButton';
