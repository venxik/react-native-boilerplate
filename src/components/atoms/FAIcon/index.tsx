import React, { memo } from 'react';

import { Icon } from 'native-base';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Ionicons from 'react-native-vector-icons/Ionicons';

import type { IIconProps } from 'native-base';

export type IFAIconProps = {
  type?: 'font-awesome' | 'ionicons' | 'feather';
} & IIconProps;

export const FAIcon = memo((props: IFAIconProps) => {
  const { testID, type, ...baseProps } = props;

  let iconType: typeof FontAwesome;

  switch (type) {
    case 'ionicons':
      iconType = Ionicons;
      break;
    case 'feather':
      iconType = Feather;
      break;

    default:
      iconType = FontAwesome;
      break;
  }

  return <Icon testID={testID} as={iconType} {...baseProps} />;
});
FAIcon.displayName = 'FAIcon';
