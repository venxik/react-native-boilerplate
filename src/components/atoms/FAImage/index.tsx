import React, { memo } from 'react';

import { Image } from 'native-base';

import type { IImageProps } from 'native-base';

export type IFAImageProps = IImageProps;

export const FAImage = memo((props: IFAImageProps) => {
  const { testID, ...baseProps } = props;

  return <Image testID={testID} {...baseProps} />;
});
FAImage.displayName = 'FAImage';
