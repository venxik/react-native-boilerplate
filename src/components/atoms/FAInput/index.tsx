import React from 'react';
import { memo } from 'react';

import { Input, InputGroup, InputLeftAddon, InputRightAddon } from 'native-base';

import type { IInputProps } from 'native-base';

export type IFAInputProps = {
  leftAddon?: string | JSX.Element;
  rightAddon?: string | JSX.Element;
} & IInputProps;

export const FAInput = memo((props: IFAInputProps) => {
  const { testID, leftAddon, rightAddon, ...baseProps } = props;

  if (leftAddon || rightAddon) {
    return (
      <InputGroup>
        {leftAddon && <InputLeftAddon children={leftAddon} />}
        <Input testID={testID} {...baseProps} placeholder="nativebase" />
        {rightAddon && <InputRightAddon children={rightAddon} />}
      </InputGroup>
    );
  }

  return <Input testID={testID} {...baseProps} />;
});
FAInput.displayName = 'FAInput';
