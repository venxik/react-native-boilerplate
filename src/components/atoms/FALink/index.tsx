import React, { memo } from 'react';

import { Link } from 'native-base';

import type { ILinkProps } from 'native-base';

export type IFALinkProps = ILinkProps;

export const FALink = memo((props: IFALinkProps) => {
  const { testID, ...baseProps } = props;

  return <Link testID={testID} {...baseProps} />;
});
FALink.displayName = 'FALink';
