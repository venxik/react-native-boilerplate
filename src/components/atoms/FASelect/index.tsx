import React, { memo } from 'react';

import { Select } from 'native-base';

import type { ISelectProps } from 'native-base';

export type IFASelectProps = {
  data: Array<{
    label: string;
    value: string;
  }>;
} & ISelectProps;

export const FASelect = memo((props: IFASelectProps) => {
  const { testID, data, ...baseProps } = props;

  return (
    <Select testID={testID} {...baseProps}>
      {data.map((item) => (
        <Select.Item label={item.label} value={item.value} />
      ))}
    </Select>
  );
});
FASelect.displayName = 'FASelect';
