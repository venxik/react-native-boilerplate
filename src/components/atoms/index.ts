export * from './FABadge';
export * from './FAButton';
export * from './FAIcon';
export * from './FAImage';
export * from './FAInput';
export * from './FALink';
export * from './FARadio';
export * from './FASelect';
export * from './FASpinner';
export * from './FASwitch';
export * from './FATextArea';
