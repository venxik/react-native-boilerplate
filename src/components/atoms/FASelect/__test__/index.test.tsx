import React from 'react';

import { FASelect } from '@components/atoms';
import { render } from '@mocks';

describe('FASelect', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<FASelect data={[]} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with children prop', () => {
    const { getByTestId } = render(
      <FASelect
        testID="test-select"
        data={[
          {
            value: '1',
            label: 'UX Research',
          },
          {
            value: '2',
            label: 'Web Development',
          },
          {
            value: '1',
            label: 'UX Research',
          },
        ]}
      />,
    );

    expect(getByTestId('test-select')).toBeTruthy();
  });
});
