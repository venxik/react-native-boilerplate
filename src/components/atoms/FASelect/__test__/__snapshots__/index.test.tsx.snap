// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FASelect should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={Object {}}
  >
    <View
      dataSet={Object {}}
      isFocused={false}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#6287A2",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TextInput
        accessible={true}
        aria-hidden={true}
        dataSet={Object {}}
        editable={false}
        focusable={false}
        importantForAccessibility="no"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholderTextColor="#a3a3a3"
        pointerEvents="none"
        secureTextEntry={false}
        selection={
          Object {
            "start": 0,
          }
        }
        selectionColor="#1f2937"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#6287A2",
            "flex": 1,
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "height": "100%",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
            "width": "100%",
          }
        }
        value=""
      />
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <RNSVGSvgView
        accessibilityRole="image"
        align="xMidYMid"
        bbHeight={16}
        bbWidth={16}
        color={-5058362}
        dataSet={Object {}}
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke=""
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "color": "#B2D0C6",
              "height": 16,
              "marginRight": 12,
              "width": 16,
            },
            Object {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor={-5058362}
        vbHeight={24}
        vbWidth={24}
      >
        <RNSVGGroup
          propList={
            Array [
              "stroke",
            ]
          }
          stroke={null}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M20.2286 6L11.9973 14.3785L3.76862 6.00268L2 7.80293L12 18L22 7.80293L20.2286 6Z"
              fill={
                Array [
                  2,
                ]
              }
              propList={
                Array [
                  "fill",
                  "stroke",
                ]
              }
              stroke={null}
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
</RNCSafeAreaProvider>
`;
